src/hotspot/share/gc/g1/g1ParScanThreadState.cpp

```diff
@@ -85,6 +85,9 @@ G1ParScanThreadState::G1ParScanThreadState(G1CollectedHeap* g1h,
     _num_optional_regions(optional_cset_length),
     _numa(g1h->numa()),
     _obj_alloc_stat(NULL),
+#ifndef PRODUCT
+    _evac_failure_inject_counter(0),
+#endif
```

Comment:
This might be less cluttering as `NOT_PRODUCT(_evac_failure_inject_counter(0) COMMA)`.  Your call...